frame/grandpa/src/lib.rs

```diff
@@ -174,15 +174,25 @@ decl_storage! {
 		/// in the "set" of Grandpa validators from genesis.
 		CurrentSetId get(fn current_set_id) build(|_| fg_primitives::SetId::default()): SetId;
 
-		/// A mapping from grandpa set ID to the index of the *most recent* session for which its members were responsible.
-		SetIdSession get(fn session_for_set): map hasher(blake2_256) SetId => Option<SessionIndex>;
+		/// A mapping from grandpa set ID to the index of the *most recent* session for which its
+		/// members were responsible.
+		SetIdSession get(fn session_for_set): map hasher(twox_64_concat) SetId => Option<SessionIndex>;
 	}
 	add_extra_genesis {
 		config(authorities): AuthorityList;
 		build(|config| Module::<T>::initialize_authorities(&config.authorities))
 	}
 }
 
+mod migration {
+	use super::*;
+	pub fn migrate<T: Trait>() {
+		for i in 0..=CurrentSetId::get() {
+			SetIdSession::migrate_key_from_blake(i);
```

Comment:
@rphmeier please confirm this is correct.

Reply:
Looks fine to me